sources/views/contact-info.js

```diff
@@ -52,31 +66,81 @@ export default class ContactInfo extends JetView {
 			</div>
 			<div><span class="user-info--status">${contact.Status || "no data"}</span></div>
 		</div>
-		</div>`;
-			}
+		</div>`
 		};
 
-		const view = { rows: [formButtons, userInfo] };
+		const bottomTabbar = {
+			rows: [
+				{
+					view: "tabbar",
+					localId: "contactsData",
+					value: "Activities",
+					options: [{value: "Activities"}, {value: "Files"}]
+				},
+				{
+					cells: [
+						{localId: "Activities", rows: [ActivitiesOfContact]},
+						{localId: "Files", rows: [FilesOfContact]}
+					]
+				}
+			]
+		};
+
+		const view = {rows: [formButtons, userInfo, bottomTabbar]};
 
 		return view;
 	}
 
 	init() {
 		this.info = this.$$("userInfo");
+		this.$$("contactsData").attachEvent("onChange", value => this.$$(value).show());
 	}
-	urlChange(view, url) {
-		const elementId = this.getParam("id");
+
+	urlChange() {
+		const elementId = this.getParam("id", true);
 
 		statuses.waitData.then(() => {
-			if (contacts.exists(elementId)) {
-				const contact = webix.copy(contacts.getItem(elementId));
-				const statusId = contact.StatusID;
+			if (!contacts.exists(elementId)) {
+				return;
+			}
 
-				if (statuses.exists(statusId)) {
-					contact.Status = statuses.getItem(statusId).Value;
-				}
-				this.info.setValues(contact);
+			const contact = webix.copy(contacts.getItem(elementId));
+			const statusId = contact.StatusID;
+			contact.Birthday = webix.i18n.longDateFormatStr(contact.Birthday);
+			if (statuses.exists(statusId)) {
+				contact.Status = statuses.getItem(statusId).Value;
+			}
+			this.info.setValues(contact);
+		});
+	}
+
+	confirmDeleteContact() {
+		const id = this.getParam("id", true);
+		if (!id && !contacts.exists(id)) {
+			return;
+		}
+		this.webix
+			.confirm({
+				type: "confirm-warning",
+				text: "Are you sure?"
+			})
+			.then(() => {
+				this.deleteContact(id);
+			});
+	}
+
+	deleteContact(id) {
+		activities.serialize().forEach((item) => {
```

Comment:
Instead of `serialize` you can [find](https://docs.webix.com/api__link__datacollection_find.html) all activities with needed `ContactId` (method will return array of matching objects). It would be more clear implementation.

Note also that [remove](https://docs.webix.com/api__link__datacollection_remove.html) can accept a non-empty array (and should not be called with an empty array or undefined value). 
You can do something like https://snippet.webix.com/2jc3xtzw

Reply:
Fixed. But i will'not use map because it will add an extra operation.

Reply:
Separate `activities.remove` for each item will trigger several iterations of data removing (which is a complex process), so `map` is the lesser evil.

Reply:
Sorry, I didn't understand that we can pass array in remove.
Fixed.